docs/pages/reference/architecture/architecture.mdx

```diff
@@ -1,10 +1,125 @@
 ---
-title: Teleport Architecture Guides
-description: Get detailed information about how Teleport works
+title: Teleport Architecture
+description: Provides detailed information about how Teleport works.
 ---
 
-In this section, you will find detailed information about Teleport's internal
-architecture. Read these guides if you are interested in learning how Teleport
-works.
+This guide explains the technical architecture of Teleport. Before reading this
+guide, we recommend that you read the [Core Concepts](../../core-concepts.mdx)
+page, which describes the components of a Teleport cluster.
 
-(!toc!)
+## Teleport control plane
+
+The Teleport **control plane** consists of the Teleport Auth Service and
+Teleport Proxy Service. On Teleport Enterprise (Cloud), the control plane is
+fully managed on Teleport infrastructure. Read about [Teleport Enterprise
+(Cloud) Architecture](teleport-cloud-architecture.mdx).
+
+### Teleport Auth Service
+
+The Teleport Auth Service performs three main functions:
+- **Maintains certificate authorities** that sign host and client certificates
+  for components of your Teleport cluster as well as (for certain self-hosted
+  resources) your own infrastructure. 
+- **Stores cluster configurations** as dynamic resources, including roles, local
+  users, and certain kinds of Teleport-protected infrastructure resources.
+- **Collects cluster data** such as audit events and session recordings.
+
+Cluster components communicate with the Auth Service to manage certificates,
+dynamic resources, audit events, and session recordings through a [gRPC
+API](api-architecture.mdx).
+
+For more information about the Teleport Auth Service, read the following guides:
+
+- [Authentication](authentication.mdx)
+- [Authorization](authorization.mdx)
+
+### Teleport Proxy Service
+
+The Teleport Proxy Service enables components in a Teleport cluster to
+communicate securely with the Teleport Auth Service. With the Proxy Service,
+users can use the public internet to access infrastructure in private networks.
+
+The Proxy Service implements an SSH server. Teleport Agents establish reverse
+tunnels with the SSH server to receive traffic from (and return traffic to)
+Teleport users. Auth Service clients like the Event Handler and Access Request
+plugins also route traffic through the Proxy Service's SSH server, and
+authenticate to the Proxy Service with an SSH client certificate.
+
+The Proxy Service serves the Teleport Web UI, which in Teleport Enterprise
+(Cloud), is available at the address of your Teleport account (e.g.,
+`example.teleport.sh`).
+
+Read more about [Teleport Proxy Service Architecture](proxy.mdx). You can also
+read about the architecture of Teleport Proxy Service features:
+
+- [TLS Routing](tls-routing.mdx)
+- [Proxy Peering](proxy-peering.mdx)
+
+## Enrolling resources
+
```

Comment:
This is a little out of scope, but it seems odd we don't have an overview of enrolling resources, like a general intro page here: https://goteleport.com/docs/enroll-resources

We have enrolling apps, k8s clusters, DBs etc but they're all rather distinct

Reply:
Yeah, that's a good point. I quickly crammed these all into the same section when [reorganizing the docs](https://github.com/gravitational/teleport/pull/44327) to tidy up the sidebar, but didn't add an introduction to the section. I can open an issue